lib/Jit/LLILCJit.cpp

```diff
@@ -394,3 +395,107 @@ void __cdecl LLILCJit::fatal(int Errnum, ...) {
 void LLILCJit::signalHandler(void *Cookie) {
   // do nothing
 }
+
+template <typename ObjSetT, typename LoadResult>
```

Comment:
Only if they're intended for use in more than one translation unit. In this case, the function is local to LLILCJit.cpp, as so does not need to be defined in the header file. It might be better, in fact, if `ObjectLoadListener` was simply a forward declaration in LLILCJit.h and defined completely in LLILCJit.cpp.